pkg/ext-proc/health.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	healthPb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+	"inference.networking.x-k8s.io/gateway-api-inference-extension/api/v1alpha1"
+	klog "k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type healthServer struct {
+	client.Client
+}
+
+func (s *healthServer) Check(ctx context.Context, in *healthPb.HealthCheckRequest) (*healthPb.HealthCheckResponse, error) {
```

Comment:
I think this can check that the datastore populated the inference pool instead of actually doing a pull from the server?

Reply:
Okay, based on https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/148#discussion_r1910990336 I will revert to my original approach (minus checking for InferenceModels).

Reply:
The check is now based on whether the InferencePool exists in the data store.